integrations/access/servicenow/app.go

```diff
@@ -258,27 +271,53 @@ func (a *App) onWatcherEvent(ctx context.Context, event types.Event) error {
 }
 
 func (a *App) onPendingRequest(ctx context.Context, req types.AccessRequest) error {
-	if len(req.GetSystemAnnotations()) == 0 {
-		logger.Get(ctx).Debug("Cannot proceed further. Request is missing any annotations")
-		return nil
+	log := logger.Get(ctx)
+
+	reqID := req.GetName()
+
+	resourceNames, err := a.BaseApp.GetResourceNames(ctx, req)
+	if err != nil {
+		return trace.Wrap(err)
 	}
 
-	// First, try to create a notification incident.
-	isNew, notifyErr := a.tryNotifyService(ctx, req)
+	reqData := RequestData{
+		User:              req.GetUser(),
+		Roles:             req.GetRoles(),
+		RequestReason:     req.GetRequestReason(),
+		SystemAnnotations: req.GetSystemAnnotations(),
+		Resources:         resourceNames,
+	}
+
+	// Create plugin data if it didn't exist before.
+	isNew, err := a.modifyPluginData(ctx, reqID, func(existing *PluginData) (PluginData, bool) {
+		if existing != nil {
+			return PluginData{}, false
+		}
+		return PluginData{RequestData: reqData}, true
+	})
+	if err != nil {
+		return trace.Wrap(err, "updating plugin data")
+	}
 
 	// To minimize the count of auto-approval tries, let's only attempt it only when we have just created an incident.
 	// But if there's an error, we can't really know if the incident is new or not so lets just try.
-	if !isNew && notifyErr == nil {
+	if !isNew {
 		return nil
 	}
-	// Don't show the error if the annotation is just missing.
-	if errors.Is(notifyErr, errMissingAnnotation) {
-		notifyErr = nil
+
+	log.Infof("Creating servicenow incident for : %v", reqID)
+	if err = a.createIncident(ctx,reqID, reqData); err != nil {
+		return trace.Wrap(err, "creating ServiceNow incident")
+	}
+
+	if reqReviews := req.GetReviews(); len(reqReviews) > 0 {
+		if err = a.postReviewNotes(ctx, reqID, reqReviews); err != nil {
+			return trace.Wrap(err, "posting review notes")
+		}
```

Comment:
If I understand correctly, if we already opened an incident (or tried to open an incident) `isNew` is false and we return. This means we never reach the `postReviewNotes()` if an access request gets reviewed but is not yet approved.

I might be wrong but I think we want to update the incident with the review notes when receiving the update events. If this is really a bug, it was already present before this PR but I think we could fix it here.

Reply:
We are not checking for auto-approval if there's already a review :thinking: 
It should be ok, but I was expecting to see the auto-approval logic to be applied even if there's an approval already

Eg it requires two approvals, but only one was given, and now the auto-approval can't kick in

Reply:
What would you suggest as the ideal way the plugin should handle this?
Ill update it so the postReviewNotes gets called even if the PR is not approved yet.
But for the case of access requests that require more than 1 approval should the plugin only look at the request once it only needs 1 more approval to be approved?

Reply:
Every review should be logged as a note in ServiceNow
When it tries to auto-approve, it shouldn't matter whether it has 0 reviews or N reviews, it should be approved regardless.

Does the above look correct?
Sorry, I might be overlooking something

Reply:
Currently it just approves it in the form of an access review but that will still only count as one of the required reviews. and it will do this no matter how many reviews there are already. Is this what you were thinking or should it be the case that it should fully approve the request instead of just counting as a single approval?

Reply:
I thought that our auto-approval did a complete approval, and not just an access review.
That's not the case for other plugins?

Reply:
Looking at the pagerduty auto approval it seems to do the same thing as is done here, where the plugin just submits a single approval. https://github.com/gravitational/teleport/blob/7767bf5e0ff7704b39819e4ec144af08b7557716/integrations/access/pagerduty/app.go#L498

Reply:
Thank you for looking into it.
I guess my assumption was wrong, sorry.

Reply:
If user A requests access, and it requires 3 approvals.
- The plugin creates an incident in servicenow.
- User B approves the access request. It still needs 2 approvals to be usable.
- The plugin should:
  - Update the incident message to reflect how many approvals are still required (not mandatory, not all plugins are doing this)
  - Post review notes in the incident saying User B reviewed the access request (mandatory)
  
Here, when User B approves the access request, there's already plugindata (because the AR was posted on creation) and `isNew` is false. This means the plugin will return and skip the whole "post review notes" part.

I understand we don't want to evaluate if the user is on-call for every message, only on creation, but I think the plugin must update the review notes every time. This is also what the pagerduty plugin is doing.

I suggest moving the "postReviewNotes" before the `!isNew` check.

Reply:
Ah i see, have moved the rpostReview notes